apps/custom-ui-app-docusign/server/data/connection.js

```diff
@@ -0,0 +1,90 @@
+const Account = require("./account");
+const util = require("../util/helper")
+
+Account.sync();
+
+// Get settings associated with an account
+const getSettings = async (company_id) => {
+    try {
+        const record = await Account.findOne({
+            where: {
+                company_id
+            }
+        });
+        if (record && String(record.dataValues.settings).length > 1)
+            return {
+                configured: true,
+                values: JSON.parse(record.dataValues.settings)
+            }
+        else
+            return {
+                configured: false
+            }
+    } catch (error) {
+        debug(error);
+        throw new Error("Getting app settings from database failed");
+    }
+}
+
+// Get Access Token Associated with a company
+const getToken = async (company_id) => {
+    try {
+        const record = await Account.findOne({
+            where: {
+                company_id
+            }
+        });
+        if (record && String(record.dataValues.refresh_token).length > 1) {
+            const token = await util.getNewToken(record.dataValues.refresh_token);
+            return {
```

Comment:
Do I understand correctly that it gets new token every time when this function is called? I think it should do it only if access token has expired.

Reply:
For this one, shall I add a comment / disclaimer in the code than implement time based refresh ?

Reply:
Got it.